### DISCUSSION TITLE
ggwave for VR (Unity)

### DISCUSSION THREAD

**Discussion by Dalbyte** [ggerganov/ggwave#64] [Ideas]
## ggwave for VR (Unity)

I'm really excited that the app works so well. Ultrasonic mode is magic. ❤️

### old way
 
![waverNew.jpg](https://alexanderdalbert.de/link/ggwaver/waverOld.jpg)
 
**01+02**
In VR applications you often have to log in, which is super annoying. Usually this process involves a second screen. First you receive a code after the registration, which you then have to enter on a VR keyboard. 
 
**03**
You have to lift up the VR headset look at the screen and remember the password, then switch back to enter it. 
 
A very frustrating experience. 

---
 
  ### new way
  ![waverNew.jpg](https://alexanderdalbert.de/link/ggwaver/waverNew.jpg)
 
Unfortunately, you cannot access the camera during the quest(vr heatset) for privacy reasons. (otherwise this could be solved via QR code login) 
 
But the VR application can access the microphone, if the user allows it. So with your method this could be way less frustrating.

Another example were this could be handy is if several users are in a cooperative application in the same real space and they want to enter a virtual coworking space. In most applications you have to painstakingly dictate the code to everyone until they successfully have joined the space.

With Waver you could play the sound once and all VR headsets would connect to the same virtual space. Without having to ping the network in a complicated way.

---

### my attempt

I tried it with ggwave-rx (Very basic/SDL)  to get this C++ in Unity(C#) with [Tutorials](https://www.youtube.com/watch?v=qljLhXfVt78).

Unfortunately I failed because of my skills and understanding.
Maybe someone else finds a better way.
🙂

**Comment by ggerganov** [comment]
Thank you for this post and the nice illustrations!

Your idea sounds like a great application for `ggwave`. I don't have experience with Unity, but I might give it a try at some point.
Hopefully someone with enough knowledge would come by and help with porting `ggwave` to Unity (C#).

**Reply by Dalbyte** [reply]
I hope you will succeed. I will keep trying 🙂.
